### DISCUSSION TITLE
Ice.Compression.Level

### DISCUSSION THREAD

**Discussion by yinghaoyu** [zeroc-ice/ice#1481] [Q&A] [answered]
Usually I use rpc to fetch data, my rpc function like this:
```cpp
DATA d;
opprx->rpcfun(&d);
```
Catch rpc port `1000` by:
```bash
tcpdump -nn -X port 10000 -i eth0
```
Results of package size as below,
`Ice.Compression.Level=1`:
```
53
2920
2920
2920
2920
2920
2075
```

`Ice.Compression.Level=5`:
```
53
2920
2920
2920
2920
2920
2075
```
Obviously,it does not same to be in effect. How to use it ?

**Comment by pepone** [comment, marked as answer]
Ice never compresses messages smaller than 100 bytes, the endpoint on which the message is sent has to include `-z` compression flag.

See https://doc.zeroc.com/ice/3.7/ice-protocol-and-encoding/protocol-compression

**Reply by yinghaoyu** [reply]
Thanks, I have including `-z`,  and it works.

But  `Ice.Compression.Level=1` and `Ice.Compression.Level=5` are equivalent in terms of my catch result，is this config deprecated?

**Reply by pepone** [reply]
What Ice version and language are you using?

**Reply by yinghaoyu** [reply]
`Ice 3.3.1`(https://zeroc.com/downloads/archive/ice/3.3.1) in `Cpp` project, truly it's outdated.
